### DISCUSSION TITLE
Question: fnf modding errors

### DISCUSSION THREAD

**Discussion by ComicHB** [FunkinCrew/Funkin#1573] [Compiling help]
Please help me I've been working on this for over a week (mostly due to errors)
I've got the APIstuff folder 

C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
Sys_error("export/debug/windows/obj/Options.txt: Permission denied")
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command lime build windows -debug --haxelib=hxcpp-debug-server --connect 6000" terminated with exit code: 1.
I use visual studio code

**Comment by khi-ayo** [comment]
I think I saw this asked before in a different forum thing, and someone replied that it still works even if u ignore the warning. I get told the same thing too btw and I think the game runs alright?

**Comment by ComicHB** [comment]
@khi-ayo 

The issue is that it stops my game from compiling

**Comment by bonkmaykr** [comment]
Let's help break this down so you can understand what's happening.
```
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 10-23 : Warning : This typedef is deprecated in favor of haxe.xml.Access
C:/HaxeToolkit/haxe/lib/polymod/git/polymod/format/XMLMerge.hx:189: characters 30-43 : Warning : This typedef is deprecated in favor of haxe.xml.Access
```
This part isn't too important. "deprecated" just means that that code is no longer recommended or supported and parts of it are obsolete, but the code will still work because those old features haven't been removed yet. The warning isn't really anything terrible or urgent, it's just there to let you know incase you decide to move to a newer Haxe version in the future.

```
Sys_error("export/debug/windows/obj/Options.txt: Permission denied")
The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command lime build windows -debug --haxelib=hxcpp-debug-server --connect 6000" terminated with exit code: 1.
```
This is the part where the problem lies.
It seems like whatever the compiler was trying to do, your OS stopped it.

"exit code: 1" just means a standard crash.

Try running as administrator if possible. The administrator command prompt can be found by pressing Win + X and then `Command Prompt (Administrator)` or `Powershell (Administrator)`. If you're using Powershell you'll have to run EXEs by preceeding them with "./".

**Comment by ComicHB** [comment]
@bonkmaykrQ 

So If i do that does it run everything as administrator? I have to F5 the command to run and build ? Also whenever the game did lunch it wouldn't build an .exe just lunch the game ? Would you know why it does so? 

Thank you so much for trying to help 

Ok so i tried it, the commands you have me doesn't seem to work on visual studio code, and i don't have an exe of the game so i can't really do anyways, maybe j don't understand what you're telling me and I'm sorry about that, my discord is ComicHB#7984 if anyone wanna add me and explain to me how to fix this

**Comment by bonkmaykr** [comment]
I don't use Haxe so I'm not sure where you're compiling from, I assumed you were doing so from cmd. If that's not the case, you could try finding the compiler's executable, opening Properties > Compatibility and then enabling Run as administrator. 

I use VS2019 and I'd recommend you do so as well as VS Code is for more lightweight projects.

> the commands you have me doesn't seem to work on visual studio code

I didn't provide any commands, I'm not sure what you are talking about.

**Comment by ghost** [comment]
I have a feeling that @ComicHB is using lime test html5 -debig because of the HTTP server.

**Comment by ComicHB** [comment]
@VMGuy23  i am using lime html5 indeed, I'm trying windows -debug final and it has been building for like 4-5 hors now and it opened the game (still black screen but at least something worked) so I'm waiting for now

**Comment by ComicHB** [comment]
@bonkmaykrQ  sorry u meant the console doesn't open with your command, that's what I meant
